en/developer_guide/api_transformer.rst

```diff
@@ -0,0 +1,133 @@
+we have decide to use the `Facade design pattern`_.
+
+If you want to add a new entity to the project, and you will expose it through the API,
+an easy way would be to directly serialize the object to render it.
+
+This solution is not ideal for a main reason: the response is going to move each
+time you will modify your model.
+
+Open Orchestra solution
+-----------------------
+
+Open Orchestra gives you the opportunity to decouple the data exposition from the
+data storage.
+
+To simplify this documentation, let's say that we want to expose a ``Foo`` document.
+
+.. code-block:: php
+
+    class Foo
+    {
+        /**
+         * @var string
+         */
+        public $bar;
+    }
+
+Facade
+~~~~~~
+
+The facade is the object that will be exposed through the API, let's also say that you
+need to put the value of ``Foo::bar`` inside a ``baz`` propety.
+
+Create the ``FooFacade`` with the ``baz`` property:
+
+.. code-block:: php
+
+    namespace FooBundle\Facade;
+
+    use JMS\Serializer\Annotation as Serializer;
+    use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+
+    class FooFacade implements FacadeInterface
+    {
+        /**
+         * @Serializer\Type("string")
+         */
+        public $baz;
+    }
+
+The ``FooFacade`` class should implement the ``FacadeInterface`` to be reconized as
+a facade in the rest of the application.
+
+Transformer
+~~~~~~~~~~~
+
+As the transformation for the ``bar`` to ``baz`` property is not intuitive, you will
+need to create a Transformer to perform the modification.
+
+Create the ``FooTransformer``:
+
+ * It must implement the ``OpenOrchestra\BaseApi\Transformer\TransformerInterface``
+ * Be registered with the tag ``open_orchestra_api.transformer.strategy``
+
+.. code-block:: php
+
+    namespace FooBundle\Transformer;
+
+    use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+    use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
+    use FooBundle\Facade\FooFacade;
+
+    class FooTransformer extends AbstractTransformer
+    {
+        public function getName()
+        {
+            return 'foo';
+        }
+
+        public function tranform($foo)
+        {
+            $facade = new FooFacade();
+            $facade->baz = $foo->bar;
+
+            return $facade;
+        }
+    }
+
+And the declaration :
+
+.. code-block:: yaml
+
+    foo.transformer.foo:
+        class: FooBundle\Transformer\FooTransformer
+        tags:
+            - { name: open_orchestra_api.transformer.strategy }
+
+To limit the dependancy injection, we have create a ``TransformerManager`` which will
+known all the transformers. This way, you will be able to directly call an other
+transformer with the method ``$this->getTransformer('foo')``.
+
+You will also have access to the ``Router`` and to the ``GroupContext``
+(see `group context page`_)
+
+Usage
+~~~~~
+
+In the controller, to use the transformer, you need to get it through the
```

Comment:
Maybe something more fluent like 'In the controller, you can accesss your transformer via the TransformerManager'